executor/show.go

```diff
@@ -329,6 +330,20 @@ func (e *ShowExec) fetchShowColumns() error {
 	}
 
 	cols := tb.Cols()
+	if tb.Meta().IsView() {
+		planBuilder := plannercore.NewPlanBuilder(e.ctx, e.is)
+		viewLogicalPlan, err := planBuilder.BuildDataSourceFromView(e.DBName, tb.Meta())
+		if err != nil {
+			return err
+		}
+		viewSchema := viewLogicalPlan.Schema()
+		for _, col := range cols {
+			viewColumn := viewSchema.FindColumnByName(col.Name.L)
+			if viewColumn != nil {
+				col.FieldType = *viewColumn.GetType()
```

Comment:
We can't get the column type from the columns returned by `tb.Cols()`?

Reply:
Yes, view's column tableinfo only store column name, so we need to extract sql's schema.column to get column type.

Reply:
Can we fill the column type info when this table info is generated?

Reply:
I don't think so, because undertable's column type and column length can be change.

Reply:
```sql
MySQL(root@localhost:test) > create table t(a bigint, b bigint);
Query OK, 0 rows affected (0.04 sec)

MySQL(root@localhost:test) > create view v as select * from t;
Query OK, 0 rows affected (0.07 sec)

MySQL(root@localhost:test) > show columns from v;
+-------+------------+------+-----+---------+-------+
| Field | Type       | Null | Key | Default | Extra |
+-------+------------+------+-----+---------+-------+
| a     | bigint(20) | YES  |     | NULL    |       |
| b     | bigint(20) | YES  |     | NULL    |       |
+-------+------------+------+-----+---------+-------+
2 rows in set (0.01 sec)

MySQL(root@localhost:test) > drop table t;
Query OK, 0 rows affected (0.03 sec)

MySQL(root@localhost:test) > create table t(a double, b double, c double);
Query OK, 0 rows affected (0.07 sec)

MySQL(root@localhost:test) > show columns from v;
+-------+--------+------+-----+---------+-------+
| Field | Type   | Null | Key | Default | Extra |
+-------+--------+------+-----+---------+-------+
| a     | double | YES  |     | NULL    |       |
| b     | double | YES  |     | NULL    |       |
+-------+--------+------+-----+---------+-------+
2 rows in set (0.01 sec)
```

Got it. Confirmed in MySQL.

Reply:
@AndrewDi Could you add a comment to specify why we need to build logical plan on the view to get the column types?

Reply:
Done